virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -160,7 +160,8 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
             this.switchWindow(1);
         }
         else if (event.which === KeyCode.KEY_W) {
-          if(this.focusedWindow !== null) {
+          // We should not be able to close a standalone mode application window
+          if(this.focusedWindow !== null && !this.focusedWindow.isFullscreenStandalone) {
```

Comment:
What if `focusedWindow` is `undefined`? `undefined !== null` is `true`.

Reply:
this.focusedWindow is instantiated as null, so it will always be an object or null (there's a dozen other comparisons like this) but it could lead to missed edgecases in the future so I'll change them all